### PR TITLE
Container not shut down on task creation failure

### PR DESCRIPTION
**Pull request by mfranciszkiewicz**
- shuts down the container if no more tasks are pending in order to free resources (i.e. bound ports)
- changes `RequestedTask`'s status to `errorCreating` on failure

### CODE REVIEW

File: golem/task/rpc.py [hunk 1]
```diff
@@ -529,13 +531,14 @@ def _create_legacy_task(
...
-        return task_id
+        return task_id, None
 
+    @defer.inlineCallbacks
     def _create_task_api_task(
             self,
             golem_params: dict,
             app_params: dict,
-    ) -> TaskId:
+    ):
```
Comment (maaktweluit) [inline golem/task/rpc.py:541]: this still returns a `TaskId`

**Review by maaktweluit — CHANGES_REQUESTED** [review]
LGTM!

One small return type comment

File: golem/task/rpc.py (hunk 1 above)
Comment (mfranciszkiewicz) [reply golem/task/rpc.py:541]: The type returned is not `TaskId` but a `Generator` generic type due to `inlineCallbacks`

Comment (mfranciszkiewicz) [reply golem/task/rpc.py:541]: I have issues with various linters regarding the returned type when I try to hint it as `Generator[...]` - I haven't seen anyone getting it right

Comment (maaktweluit) [reply golem/task/rpc.py:541]: ahh, damn linters :(

ok, lets leave it like this then :)

**Review by maaktweluit — APPROVED** [review]

File: golem/task/requestedtaskmanager.py [hunk 2]
```diff
@@ -265,6 +282,32 @@ def create_task(
...
+                task = RequestedTask.get(task_id=task_id)
+                task.status = TaskStatus.errorCreating
+                task.save()
+                if not app_id:
+                    app_id = task.app_id
+            except RequestedTask.DoesNotExist:
+                pass
+            try:
+                if app_id:
+                    await self._shutdown_app_client(app_id)
```
Comment (Wiezzel) [inline golem/task/requestedtaskmanager.py:305]: Shouldn't we check here if there are no more tasks running on this client?

Comment (mfranciszkiewicz) [reply golem/task/requestedtaskmanager.py:305]: The check is inside `_shutdown_app_client`

Comment (Wiezzel) [reply golem/task/requestedtaskmanager.py:305]: Oh, of course.

**Review by Wiezzel — APPROVED** [review]

**Pull request merged**